crates/top/re_sdk/src/lenses.rs

```diff
@@ -0,0 +1,745 @@
+            Self::Func(func) => func(list_array),
+        }
+    }
+}
+/// Private module with the AST-like definitions of lenses.
+///
+/// **Note**: We should not leak those into the public API interface,
+/// so that we can evolve the definition of lenses over time, if requirements
+/// change.
+mod ast {
+    use super::{
+        ComponentDescriptor, ComponentIdentifier, EntityPath, Op, ResolvedEntityPathFilter,
+    };
+
+    pub(super) struct InputColumn {
+        pub entity_path_filter: ResolvedEntityPathFilter,
+        pub component: ComponentIdentifier,
+    }
+
+    pub(super) struct OutputColumn {
+        pub entity_path: EntityPath,
+        pub component_descr: ComponentDescriptor,
+        pub ops: Vec<Op>,
+        // TODO(grtlr): It would be much nicer if static could be infered from the output of the operations?
+        pub is_static: bool,
+    }
+}
+
+/// Provides convenient function to create a [`Lens`].
+pub struct LensBuilder(Lens);
+
+impl LensBuilder {
+    /// The column on which this [`Lens`] will operate on.
+    pub fn input_column(
+        entity_path_filter: EntityPathFilter,
+        component: impl Into<ComponentIdentifier>,
+    ) -> Self {
+        Self(Lens {
+            input: ast::InputColumn {
+                entity_path_filter: entity_path_filter.resolve_without_substitutions(),
+                component: component.into(),
+            },
+            outputs: vec![],
+        })
+    }
+
+    /// Can be used to define one or more output columns that are derived from the [`Self::input_column`].
+    pub fn output_column(
+        mut self,
+        entity_path: impl Into<EntityPath>,
+        component_descr: ComponentDescriptor,
+        ops: impl IntoIterator<Item = Op>,
+    ) -> Self {
+        let column = ast::OutputColumn {
+            entity_path: entity_path.into(),
+            component_descr,
+            ops: ops.into_iter().collect(),
+            is_static: false,
+        };
+        self.0.outputs.push(column);
+        self
+    }
+
+    /// Can be used to define one or more output columns that are derived from the [`Self::input_column`].
+    pub fn static_output_column(
+        mut self,
+        entity_path: impl Into<EntityPath>,
+        component_descr: ComponentDescriptor,
+        ops: impl IntoIterator<Item = Op>,
+    ) -> Self {
+        let column = ast::OutputColumn {
+            entity_path: entity_path.into(),
+            component_descr,
+            ops: ops.into_iter().collect(),
+            is_static: true,
+        };
+        self.0.outputs.push(column);
+        self
+    }
+
+    /// Finalizes this builder and returns the corresponding lens.
+    pub fn build(self) -> Lens {
+        self.0
+    }
+}
+
+/// A lens that transforms component data from one form to another.
+///
+/// Lenses allow you to extract, transform, and restructure component data
+/// as it flows through the logging pipeline. They are applied to chunks
```

Comment:
"logging" sounds like the `send_` APIs don't affect it.
Also, here we don't know how it's gonna be used